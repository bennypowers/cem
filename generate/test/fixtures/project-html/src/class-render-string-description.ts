@customElement('class-render-string-description')
class ClassRenderStringDescription extends LitElement {
  render() {
    return html`
      <!-- anonymous -->
      <slot></slot>
      <!-- name="slot" -->
      <slot name="slot"></slot>
      <!-- name="part" -->
      <a part="part"></a>
    `;
  }
}
