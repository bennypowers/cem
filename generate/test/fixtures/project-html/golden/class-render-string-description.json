{
  "schemaVersion": "2.1.0",
  "modules": [
    {
      "kind": "javascript-module",
      "path": "src/class-render-string-description.js",
      "declarations": [
        {
          "name": "ClassRenderStringDescription",
          "superclass": {
            "name": "LitElement",
            "package": "lit"
          },
          "source": {
            "href": "https://github.com/bennypowers/cem/tree/main/generate/test/fixtures/project-html/src/class-render-string-description.ts#L1"
          },
          "kind": "class",
          "tagName": "class-render-string-description",
          "slots": [
            {
              "name": "",
              "description": "anonymous"
            },
            {
              "name": "slot",
              "description": "name=\"slot\""
            }
          ],
          "cssParts": [
            {
              "name": "part",
              "description": "name=\"part\""
            }
          ],
          "customElement": true
        }
      ],
      "exports": [
        {
          "kind": "custom-element-definition",
          "name": "class-render-string-description",
          "declaration": {
            "name": "ClassRenderStringDescription",
            "module": "src/class-render-string-description.js"
          }
        }
      ]
    }
  ]
}